OpenSim/Actuators/VandenBogert2011Muscle.cpp

```diff
@@ -0,0 +1,1234 @@
+            muscleLength, projFibLenNorm, activ, projFibVelNorm_guess,
+            activdot_guess, u, 0);
+
+    return Results;
+}
+
+
+//------------------------------------------------------------------------------
+VandenBogert2011Muscle::ImplicitResidual
+VandenBogert2011Muscle::calcImplicitResidual(SimTK::Vec2 y,
+                                             SimTK::Vec2 ydot_guess,
+                                             double muscleLength,
+                                             double u, int returnJacobians=0)
+const
+{
+    // Overload method for state vectors as parameters
+    VandenBogert2011Muscle::ImplicitResidual Results= calcImplicitResidual(
+            muscleLength, y[0], y[1], ydot_guess[0], ydot_guess[1], u,
+            returnJacobians);
+    return Results;
+
+}
+
+//------------------------------------------------------------------------------
+VandenBogert2011Muscle::ImplicitResidual VandenBogert2011Muscle::
+calcImplicitResidual(double muscleLength, double projFibLenNorm, double activ,
+                     double projFibVelNorm, double activdot, double u,
+                     int returnJacobians=0) const {
+
+
+    //TODO: Match symbols to doxygen diagram and Add symbolic equations comments
+    //TODO: May want to make this into a seperate function
+
+    // -------------------------Parameters----------------------------//
+    //F_{iso} Maximum isometric force that the fibers can generate
+    double maxIsoForce = getMaxIsometricForce();
+
+    //u_{max} (dimensionless) strain in the series elastic element at load of
+    //      maxIsometricForce
+    double fMaxTendonStrain = getFMaxTendonStrain(); //Strain in the series
+    //      elastic element at load of Fmax
+
+    //W (dimensionless) width parameter of the force-length relationship of
+    //      the muscle fiber
+    double fl_width = getFlWidth();
+
+    //AHill (dimensionless) Hill parameter of the force-velocity relationship
+    double fv_AHill= getFvAHill();   //Hill parameter of the f-v relationship
+
+    //FV_{max} (dimensionless) maximal eccentric force
+    double fv_maxMultiplier = getFvmaxMultiplier();
+    //Maximal eccentric force multiplier
+
+    // L_{opt}  (m) Optimal Length of Contractile Element;
+    double optFiberLength = getOptimalFiberLength();
+    //Optimal Length of Contractile Element
+
+    //b (s/m) damping coefficient of damper parallel to the fiber
+    // (normalized to maxIsometricForce)
+    double dampingCoeff = getDampingCoefficient();
+
+    //L_{slack,fiber} Slack length of the parallel elastic element, divided by
+    //          Lceopt
+    double fiberSlackLengthNorm= getNormFiberSlackLength();
+
+    //L_{slack,tendon} (m) slack length of the tendon
+    double tendonSlackLength = getTendonSlackLength();
+
+    //T_{act} (s) Activation time
+    double activTimeConstant = getActivTimeConstant();
+
+    //T_{deact} (s) Deactivation time
+    double deactivTimeConstant = getDeactivTimeConstant();
```

Comment:
If this method stays, it should be `getDeactivationTimeConstant()` to be consistent with the same method in other Muscle classes.